File: 0462.最少移动次数使数组元素相等II/0462-最少移动次数使数组元素相等II.go
func minMoves2(nums []int) int {
    moves := 0
    //https://pkg.go.dev/sort
    sort.Ints(nums)
    mid := nums[len(nums)/2]
    for _,num := range nums {
        moves += abs( num - mid )
    }

    return moves
}

func abs(a int) int {
    if a < 0 {
        return 0 - a
    }
    return a
}

File: 0462.最少移动次数使数组元素相等II/0462-最少移动次数使数组元素相等II.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
        sort( nums.begin(), nums.end() );
        int mid = nums[nums.size()/2];
        int moves = 0;
        for( int &num : nums ) {
            moves += abs( num - mid );
        }

        return moves;
    }
};